node/node.go

```diff
@@ -546,21 +547,23 @@ func adjustNftTokenIdentifier(token string, nonce uint64) string {
 	return formattedTokenIdentifier
 }
 
-func (n *Node) getAccountHandlerAPIAccounts(address string) (state.UserAccountHandler, error) {
-	componentsNotInitialized := check.IfNil(n.coreComponents.AddressPubKeyConverter()) ||
-		check.IfNil(n.stateComponents.AccountsAdapterAPI())
+func (n *Node) decodeAddressToPubKey(address string) ([]byte, error) {
+	// Question for review: do we still require the special lazy initialization logic?
+	// > other functions seem to not care about this (and directly use the converter). E.g. GetESDTsRoles()
```

Comment:
you can remove it. maybe add a TODO and a jira task. we might as well make the node struct the standard way inside the project: via constructor that guarantees a coherent object

Reply:
Added task (EN-12399).